java/src/jmri/jmrit/display/switchboardEditor/BeanSwitch.java

```diff
@@ -908,23 +911,25 @@ protected void connectNew() {
             switchConnect.setOK(); // activate OK button on Add new device pane
             addFrame.add(switchConnect);
         }
-        addFrame.pack();
-        addFrame.setLocationRelativeTo(this);
-        addFrame.setVisible(true);
+        ThreadingUtil.runOnGUI( () -> {
+            addFrame.pack();
+            addFrame.setLocationRelativeTo(this);
+            addFrame.setVisible(true);
+        });
     }
 
     protected void cancelAddPressed(ActionEvent e) {
         if (addFrame != null) {
-            addFrame.setVisible(false);
+            ThreadingUtil.runOnGUI( () -> addFrame.setVisible(false) );
             addFrame.dispose();
-            addFrame = null;
+           addFrame = null;
```

Comment:
This change seems unintentional. The indentation is changed.

Reply:
Thanks, good spot.